### PR TITLE
Fixed bug with i18n and session locale was added

### PR DESCRIPTION
**Pull request by Axedez**
[Trello Task](https://trello.com/c/26guCa4U/64-fix-chosen-language-to-be-active-all-over-the-site)

**CHECK LIST**
- [ ] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

File: app/views/shared/account/_header.html.slim [hunk 1]
```diff
@@ -45,5 +45,5 @@ nav.navbar.navbar-expand-lg.navbar-dark.navbar-header
       li
         = link_to  destroy_user_session_path, method: :delete
           span
-            button.btn.btn-transparent.auth-btn.ml-2 
+            button.btn.btn-transparent.auth-btn.ml-2
               = t('.btn_destoy_auth')
```
Comment (mmarusyk) [inline app/views/shared/account/_header.html.slim:49]: add newline, pls

**Review by mmarusyk — APPROVED** [review]

File: app/views/shared/account/_header.html.slim (hunk 1 above)
Comment (Axedez) [reply app/views/shared/account/_header.html.slim:49]: image

Comment (Axedez) [reply app/views/shared/account/_header.html.slim:49]: Ok

**Review by marikrebega — APPROVED** [review]

**Review by matsevich — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

**Review by Swooty18 — APPROVED** [review]

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -2,7 +2,10 @@ class ApplicationController < ActionController::Base
   before_action :configure_permitted_parameters, if: :devise_controller?
   around_action :switch_locale
   layout :choose_layout
-  before_action :set_raven_context
+  before_action :set_raven_context, :set_locale
+  after_action :set_locale
```
Comment (chu4) [inline app/controllers/application_controller.rb:6]: why do you need before and after action
doesn't look right
if you stored it in session it should stay there through out request

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
-  scope '(:locale)', locale: /#{I18n.available_locales.join("|")}/, defaults: { locale: I18n.default_locale } do
+  scope '(:locale)', locale: /#{I18n.available_locales.join("|")}/ do
     devise_for :users
```
Comment (BogdanBusko) [inline config/routes.rb:3]: code alignment

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -22,7 +22,10 @@
   # rspec-expectations config goes here. You can use an alternate
   # assertion/expectation library such as wrong or the stdlib/minitest
   # assertions if you prefer.
-
+  # Default locale for tests:
+  config.before(:each, type: :feature) do
+    default_url_options[:locale] = I18n.default_locale
+  end
```
Comment (BogdanBusko) [inline spec/spec_helper.rb:28]: enter after

**Review by mmarusyk — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**